esp-hal/src/soc/esp32p4/gpio.rs

```diff
@@ -0,0 +1,478 @@
+//! # GPIO configuration module (ESP32-P4)
```

Comment:
I don't think this is publically documented, it also feels chatgpt-esque and doesn't really add much compared to just reading the code - can we remove it?